src/main/java/com/exasol/adapter/document/DataProcessingPipeline.java

```diff
@@ -0,0 +1,100 @@
+package com.exasol.adapter.document;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import com.exasol.ExaConnectionInformation;
+import com.exasol.adapter.document.documentfetcher.DocumentFetcher;
+import com.exasol.adapter.document.documentfetcher.FetchedDocument;
+import com.exasol.adapter.document.mapping.SchemaMapper;
+import com.exasol.adapter.document.mapping.SchemaMappingRequest;
+import com.exasol.errorreporting.ExaError;
+import com.exasol.sql.expresion.ValueExpressionToJavaObjectConverter;
+import com.exasol.sql.expression.ValueExpression;
+
+import akka.actor.ActorSystem;
+import akka.stream.OverflowStrategy;
+
+/**
+ * This class implements the data processing int the UDF.
+ */
+public class DataProcessingPipeline {
+    private final ValueExpressionToJavaObjectConverter valueExpressionToJavaObjectConverter;
+    private final SchemaMapper schemaMapper;
+    private final PipelineMonitor pipelineMonitor;
+
+    /**
+     * Create a new instance of {@link DataProcessingPipeline}.
+     * 
+     * @param schemaMappingRequest schema mapping request
+     */
+    public DataProcessingPipeline(final SchemaMappingRequest schemaMappingRequest) {
+        this.valueExpressionToJavaObjectConverter = new ValueExpressionToJavaObjectConverter();
+        this.schemaMapper = new SchemaMapper(schemaMappingRequest);
+        this.pipelineMonitor = new PipelineMonitor();
+    }
+
+    /**
+     * Run the data processing.
+     * 
+     * @param documentFetcher       document fetcher
+     * @param connectionInformation connection information
+     * @param rowHandler            handler for the read rows
+     * @throws InterruptedException if interrupted during execution
+     */
+    public void run(final DocumentFetcher documentFetcher, final ExaConnectionInformation connectionInformation,
+            final RowHandler rowHandler) throws InterruptedException {
+        final ActorSystem system = ActorSystem.create("DataProcessingPipeline");
+        this.pipelineMonitor.start();
+        try {
+            documentFetcher.run(connectionInformation).async()//
+                    .map(this.pipelineMonitor::onEnterBuffer1)//
+                    .buffer(600, OverflowStrategy.backpressure())//
+                    .map(this::runSchemaMapping).async()//
+                    .buffer(600, OverflowStrategy.backpressure())//
```

Comment:
Extract constant.

Reply:
Can this also be user configured? 
This is going to store 600 * 10000 (chunk size) jobs in memory, depending on situation maybe makes sense to decrease buffer size.

Reply:
I will cover in https://github.com/exasol/virtual-schema-common-document/issues/67